RecoEgamma/EgammaTools/src/EgammaPCAHelper.cc

```diff
@@ -0,0 +1,342 @@
+#include "RecoEgamma/EgammaTools/interface/EgammaPCAHelper.h"
+
+#include "DataFormats/HGCRecHit/interface/HGCRecHit.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+
+// just to retrieve a "magic" number
+#include "RecoLocalCalo/HGCalRecAlgos/interface/HGCalImagingAlgo.h"
```

Comment:
which magic numbers are actually needed?
is this just HGCalImagingAlgo::maxlayer ?

git grep shows
```
RecoLocalCalo/HGCalRecAlgos/interface/HGCal3DClustering.h:  static const unsigned int lastLayerBH = 52;
RecoLocalCalo/HGCalRecAlgos/interface/HGCalDepthPreClusterer.h:  static const unsigned int lastLayerBH = 52;
RecoLocalCalo/HGCalRecAlgos/interface/HGCalImagingAlgo.h:static const unsigned int maxlayer = 52;
```

it's time to move this to a common place.
Isn't this also known from geometry?

Reply:
This info is available from RecHitTools (taken from geometry w/ easier interface):
https://github.com/cms-sw/cmssw/blob/50392d672b587a8993578cc0ead2b20b08a3beac/RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h#L52-L53
https://github.com/cms-sw/cmssw/blob/50392d672b587a8993578cc0ead2b20b08a3beac/RecoParticleFlow/PFClusterProducer/plugins/SimMappers/RealisticSimClusterMapper.cc#L68

Reply:
I'm going to leave it to HGCal geometry experts to unify this magic number, but did add a note